packages/ts-autocomplete/src/index.ts

```diff
@@ -8,66 +9,145 @@ const debugError = createDebug('ts-autocomplete:error');
+  if (parts.length === 1 && fileName.endsWith('package.json')) {
+    // special case: when it looks up this package itself it isn't going to find
+    // it in node_modules
+    return '@mongodb-js/mongodb-ts-autocomplete/package.json';
+  }
+  return parts[parts.length - 1];
+}
+
+type EncounteredPaths = {
+  getScriptSnapshot: string[];
+  fileExists: string[];
+  readFile: string[];
+};
 
 function getVirtualLanguageService(): {
   languageService: ts.LanguageService;
   updateCode: UpdateDefinitionFunction;
   listFiles: () => string[];
+  listEncounteredPaths: () => EncounteredPaths;
 } {
-  const codeHolder: Record<TypeFilename, string> = Object.create(null);
+  // as an optimization, the contents of a file can be string or true. This is
+  // because some files are only checked for existence during module resolution,
+  // but never loaded. In that case the contents is true, not a string.
+  const codeHolder: Record<TypeFilename, string | boolean> =
+    Object.create(null);
   const versions: Record<TypeFilename, number> = Object.create(null);
   const options: ts.CompilerOptions = {
     target: ts.ScriptTarget.ES2022,
     allowJs: true,
     moduleResolution: ts.ModuleResolutionKind.NodeNext,
-    types: ['node'],
-    lib: ['es2019'],
-    //typeRoots: [],
+    types: [],
+    lib: ['es2023'],
     allowImportingTsExtensions: true,
   };
 
-  const updateCode = (newDef: Record<TypeFilename, string>): void => {
+  const updateCode = (newDef: Record<TypeFilename, string | boolean>): void => {
     for (const [key, value] of Object.entries(newDef)) {
       codeHolder[key] = value;
       versions[key] = (versions[key] ?? 0) + 1;
     }
   };
 
-  const listFiles = () => {
+  const listFiles = (): string[] => {
     return Object.keys(codeHolder);
   };
 
+  const encounteredPaths: EncounteredPaths = {
+    getScriptSnapshot: [],
+    fileExists: [],
+    readFile: [],
+  };
+  const listEncounteredPaths = (): EncounteredPaths => {
+    encounteredPaths.getScriptSnapshot.sort();
+    encounteredPaths.fileExists.sort();
+    encounteredPaths.readFile.sort();
+    return encounteredPaths;
+  };
+
   const servicesHost: ts.LanguageServiceHost = {
     getScriptFileNames: () => {
       return Object.keys(codeHolder);
     },
     getScriptVersion: (fileName) => {
+      fileName = relativeNodePath(fileName);
       return (versions[fileName] ?? 1).toString();
     },
     getScriptSnapshot: (fileName) => {
+      fileName = relativeNodePath(fileName);
       if (fileName in codeHolder) {
-        return ts.ScriptSnapshot.fromString(codeHolder[fileName]);
+        return ts.ScriptSnapshot.fromString(codeHolder[fileName].toString());
       }
 
-      return ts.ScriptSnapshot.fromString(ts.sys.readFile(fileName) || '');
+      if (process.env.CI) {
+        const result = ts.ScriptSnapshot.fromString(
+          // NOTE: some files do not exist. A good example is "typescript/lib/es2023.ts"
```

Comment:
Just FYI, maybe you already know this, I think when you see stuff like just `es2023` here it means that you need to resolve it to one of those lists of other files that we're currently listing manually, I'm pretty sure there's a method for that on ts or ts.sys that you can use, but not probably something we need to spend time figuring out right now, idk